README.md

```diff
@@ -2,6 +2,138 @@
 
 This repo contains a set of opinionated react component groups used to standardize functionality and look and feel across products.  The components are based of PatternFly with some additional functionality. 
 
+## Contribution guide
+
+### Before adding a new component:
+- make sure your use case is new/complex enough to be added to this extension
+- the component should bring a value value above and beyond existing PatternFly components
+
+### To add a new component:
+1. create a folder in `src/` matching its name (for example `src/MyNewComponent`)
+2. to the new folder add a new `.tsx` file named after the component (for example `src/MyNewComponent/MyNewComponent.tsx`)
+3. to the same folder include also a `index.ts` which will export the component as a default and then all necessary interfaces
+4. don't forget to export your component using `export * from './ComponentName'` from the root `index` file located in `src/index.ts`
+
+#### Example component:
+```
+import * as React from 'react';
+import { Text } from '@patternfly/react-core';
+import { createUseStyles } from 'react-jss';
+
+// do not forget to export your component's interface
+// always place component's interface above the component itself in the code
+export interface MyNewComponentProps {
+  text: String;
+}
+
+const useStyles = createUseStyles({
+  myText: {
+    fontFamily: 'monospace',
+    fontSize: 'var(--pf-v5-global--icon--FontSize--md)',
+  },
+})
+
+// do not use named export of your component, just a default one
+const MyNewComponent: React.FunctionComponent<MyNewComponentProps> = () => {
+  const classes = useStyles();
+
+  return (
+    <Text className={classes.myText}>
+      This is my new reusable component
+    </Text>
+  );
+};
+
+export default MyNewComponent;
+``` 
+
+#### Index file example:
+```
+export { default } from './MyNewComponent';
+export * from './MyNewComponent';
+``` 
+
+#### Component directory structure example:
+```
+src
+|- MyNewComponent
+|  |- index.ts
+|  |- MyNewComponent.tsx
+|- ...
+|- ...
+|- ...
+|- index.ts       <<-- export * from './MyNewComponent';
```

Comment:
You will have to map the default export
```ts
export * from './MyNewComponent';
export { default as MyNewComponent } from './MyNewComponent';
```
The index.ts can be also generated if the convention is strictly followed and enforced